### DISCUSSION TITLE
self.log twice problem

### DISCUSSION THREAD

**Discussion by zcx-language** [Lightning-AI/pytorch-lightning#13263] [DDP / multi-GPU / multi-node]
```
        self.test_img_metric = MetricCollection([
            PeakSignalNoiseRatio(compute_on_step=False),
            SSIM(compute_on_step=False),
        ])
        self.test_msg_metric = BitAccuracy(compute_on_step=False)
        self.test_tmsg_metric = BitAccuracy(compute_on_step=False)

    def validation_epoch_end(self, outputs: Any) -> None:
        valid_img_metric = self.valid_img_metric.compute()
        valid_tmsg_metric = self.valid_tmsg_metric.compute()
        self.valid_img_metric.reset()
        self.valid_tmsg_metric.reset()

        # Used for model checkpoint callback, only save ckpts that PSNR > `33.`
        if valid_img_metric['PeakSignalNoiseRatio'] > 30:
            self.log('tbit_acc', valid_tmsg_metric, prog_bar=True, logger=False)
        else:
            self.log('tbit_acc', 0.5, prog_bar=True, logger=False)
```

There are two metrics in my work: `psnr` and `bit_acc`. When I use `ModelCheckpoint` callback which only monitor one metric, I try use above code to choose best checkpoints where psnr > 30, but I got the error below:

**You called `self.log(tbit_acc, ...)` twice in `validation_epoch_end` with different arguments. This is not allowed**

I dont not understand why `self.log` can be called twice in `if....else...` sentence.  I run it in two gpus and I kown that from #6501 

> `validation_epoch_end` will be called in every subprocesses,   Any metric will automatically synchronize between different processes whenever metric.compute() is called.

So it is the reason of `valid_img_metric['PeakSignalNoiseRatio']` get different values in different gpus caused this problem? but the metric is synchronized.

**Comment by rohitgr7** [comment]
is it possible for you to reproduce this issue??
using https://colab.research.google.com/github/Lightning-AI/lightning/blob/master/examples/pl_bug_report/bug_report_model.ipynb
